csharp/ql/src/semmle/code/csharp/controlflow/internal/Completion.qll

```diff
@@ -279,16 +264,39 @@ private class TriedControlFlowElement extends ControlFlowElement {
     result instanceof SystemOutOfMemoryExceptionClass
   }
 
-  private Assembly getCoreLib() { assemblyCompiledWithCoreLib(this.getAssembly(), result) }
+  private CoreLib getCoreLibFromACatchClause() {
+    exists(SpecificCatchClause scc | scc = try.getACatchClause() |
+      result = scc.getCaughtExceptionType().getABaseType*().getALocation()
```

Comment:
One thing that might go a bit wrong here is that you'll end up querying `ObjectType.getALocation()` which will in fact return all CoreLibs. So perhaps
```
result = scc.getCaughtExceptionType().getBaseClass*().(SystemExceptionClass).getALocation()
```

Reply:
Also `getBaseClass`

Reply:
Yes, that's a clear oversight, good spot!